common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java

```diff
@@ -610,8 +613,60 @@ public Bundle extractInteractiveRequestBundleFromResultBundle(@NonNull final Bun
+        String serializedDCFAuthResult = resultBundle.getString(AuthenticationConstants.Broker.BROKER_DCF_AUTH_RESULT);
+        if (serializedDCFAuthResult != null) {
+            AuthorizationResult authorizationResult = ObjectMapper.deserializeJsonStringToObject(serializedDCFAuthResult, MicrosoftStsAuthorizationResult.class);
+            return authorizationResult;
+        }
+
+        // DCF not supported - thrown when BrokerFlight.ENABLE_DCF_IN_BROKER is false
+        BrokerResult brokerResult = brokerResultFromBundle(resultBundle);
+        if (brokerResult.getErrorCode() != null && brokerResult.getErrorCode().equals(ErrorStrings.DEVICE_CODE_FLOW_NOT_SUPPORTED)) {
+            Logger.error(TAG, "deviceCodeFlowAuthRequest() not supported in BrokerMsalController", new ClientException(ErrorStrings.DEVICE_CODE_FLOW_NOT_SUPPORTED, "deviceCodeFlowAuthRequest() not supported in BrokerMsalController"));
+            throw new ClientException(ErrorStrings.DEVICE_CODE_FLOW_NOT_SUPPORTED, "deviceCodeFlowAuthRequest() not supported in BrokerMsalController");
+        }
+
+        throw getBaseExceptionFromBundle(resultBundle);
+    }
+
+    /**
+     * Get acquireTokenResult from resultBundle for Device Code Flow
+     * @param resultBundle The bundle to interpret
+     * @return acquireTokenResult {@link AcquireTokenResult}
+     * @throws BaseException
+     * @throws ClientException
+     */
     @NonNull
-    public AcquireTokenResult getAcquireTokenResultFromResultBundle(@NonNull final Bundle resultBundle) throws BaseException {
+    public AcquireTokenResult getDeviceCodeFlowTokenResultFromResultBundle(@NonNull final Bundle resultBundle) throws BaseException, ClientException {
+        // DCF not supported - thrown when BrokerFlight.ENABLE_DCF_IN_BROKER is false
+        BrokerResult brokerResult = brokerResultFromBundle(resultBundle);
+        if (brokerResult.getErrorCode() != null && brokerResult.getErrorCode().equals(ErrorStrings.DEVICE_CODE_FLOW_NOT_SUPPORTED)) {
+            Logger.error(TAG, "acquireDeviceCodeFlowToken() not supported in BrokerMsalController", new ClientException(ErrorStrings.DEVICE_CODE_FLOW_NOT_SUPPORTED, "deviceCodeFlowAuthRequest() not supported in BrokerMsalController"));
+            throw new ClientException(ErrorStrings.DEVICE_CODE_FLOW_NOT_SUPPORTED, "acquireDeviceCodeFlowToken() not supported in BrokerMsalController");
+        }
+
+        if (resultBundle.getBoolean(AuthenticationConstants.Broker.BROKER_REQUEST_V2_SUCCESS)) {
+            final AcquireTokenResult acquireTokenResult = new AcquireTokenResult();
+            acquireTokenResult.setLocalAuthenticationResult(authenticationResultFromBundle(resultBundle));
+            return acquireTokenResult;
+        } else if (brokerResult.getErrorCode().equals(ErrorStrings.DEVICE_CODE_FLOW_AUTHORIZATION_PENDING_ERROR_CODE)) {
+            return null;
+        }
+
+        throw getBaseExceptionFromBundle(resultBundle);
```

Comment:
Let's add some telemetry here as well to capture the error/exception

Reply:
done